inference-engine/src/vpu/graph_transformer/src/stages/split.cpp

```diff
@@ -153,18 +153,57 @@ Stage StageBuilder::addSplitStage(
         const DataVector& outputs) {
     std::vector<DimValues> offsets;
     offsets.reserve(outputs.size());
-
     DimValues curOffset({{axis, 0}});
-    for (const auto& output : outputs) {
-        offsets.emplace_back(curOffset);
-        curOffset.set(axis, curOffset[axis] + output->desc().dim(axis));
-    }
 
-    auto stage = addSplitStage(model, name, layer, std::move(offsets), input, outputs);
+    const auto haveUnusedOutput = [](const DataVector& outputs) {
+        auto predicate = [](const vpu::Data& out) {
+            return out == nullptr;
+        };
+        return std::any_of(outputs.begin(), outputs.end(), predicate);
+    };
+
+    const auto getOutAxisSizes = [](const ie::CNNLayerPtr& layer, const Dim axis) {
+        const auto outDimsSize = layer->outData[0]->getDims().size();
+        const int idx = dimToIeInd(axis, outDimsSize);
+
+        std::vector<size_t> outAxisSizes;
+        outAxisSizes.reserve(outDimsSize);
+        for (const auto& out : layer->outData) {
+            IE_ASSERT(idx <= out->getDims().size());
```

Comment:
Please replace it with `VPU_THROW_UNLESS` with some informative message.

Reply:
Changed.